src/serialization/pkl2proto.py

```diff
@@ -0,0 +1,136 @@
+import pickle
+import sys
+import numpy as np
+import font_pb2
+
+
+def openPickle(foo):
+    f = open(foo, 'rb')
+    cur = pickle.load(f)
+    return cur
+
+
+def parseDict(d, filename):
+    #this should return the np array that we will be using in the model
+    # chars = 'abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ'
+    chars = [
+        'A', 'B', 'C', 'D', 'E', 'F', 'G', 'H', 'I', 'J', 'K', 'L', 'M', 'N',
+        'O', 'P', 'Q', 'R', 'S', 'T', 'U', 'V', 'X', 'Y', 'Z', 'a', 'b', 'c',
+        'd', 'e', 'f', 'g', 'h', 'i', 'j', 'k', 'l', 'm', 'n', 'o', 'p', 'q',
+        'r', 's', 't', 'u', 'v', 'w', 'x', 'y', 'z', 'zero', 'one', 'two',
+        'three', 'four', 'five', 'six', 'seven', 'eight', 'nine', 'space',
+        'exclam', 'quotedbl', 'numbersign', 'dollar', 'percent', 'ampersand',
+        'quotesingle', 'parenleft', 'parenright', 'asterisk', 'plus', 'comma',
+        'hyphen', 'period', 'slash', 'colon', 'semicolon', 'less', 'equal',
+        'greater', 'question', 'at', 'bracketleft', 'bracketright',
+        'asciicircum', 'underscore', 'grave', 'braceleft', 'bar', 'braceright',
+        'asciitilde'
+    ]
+
+    curveMax = 80
+    #TODO add to this list so that we know what to use
+
+    final = np.ones((len(chars), curveMax, 3,
+                     2))  # 3,2 means 3 x,y coords for quadratic bezier curve
+    final = final * 999
+
+    weird = 0
+
+    numClosedCurves = []
+    for c in range(len(chars)):
+        try:
+            # this will run per glyph
+           
+            #this is part of checking max closed curves
+            for cc in d[chars[c]]:
+                numClosedCurves.append(cc)
+                # numClosedCurves w
+
+            # curvesInGlyph = 0
+            # for counter in d[chars[c]]:
+            # curvesInGlyph +=len(d[chars[c]][counter])
+
+            # # use sum to check if there are too many curves defining a glyph
+            # if curvesInGlyph < curveMax:
+            # bezCurveIndex = 0
+            # for counter in d[chars[c]]: #this will loop through the curves/counters
+            # for curve in range(len(d[chars[c]][counter])): #loop through each curve that makes up the counter
+
+            # bez = np.asarray(d[chars[c]][counter][curve])
+            # # convert to np array to easily drop it into our data structure
+            # try:
+            # if bez.shape != (3,2):
+            # # this is just in case
+            # print("weird number of coordinates appeared:",chars[c], d[chars[c]][counter][curve], bez.shape)
+            # weird +=1
+            # final[c][bezCurveIndex] = bez
+            # bezCurveIndex+=1
+            # except:
+            # print('error with wrong num coordinates in a curve')
+
+            # # pad with 999 in the curve points as the "empty" byte
+            # # TODO perhaps think about shifting everything 999 and making the off
+            # # screen stuff at 0 so that we can reduce filesize
```

Comment:
There's a lot of commented code here. Is it worth committing? If not, delete it.

Reply:
Resolved.